### PR TITLE
actions: bluetooth: Rebase before running tests

### PR DESCRIPTION
**Pull request by carlescufi**
Like in most other actions, rebase the branch on to of the latest main
branch in order to ensure that we pick up any hotfixes.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

### CODE REVIEW

File: .github/workflows/bluetooth-tests.yaml [hunk 1]
```diff
@@ -41,7 +41,13 @@ jobs:
         uses: actions/checkout@v2
 
       - name: west setup
+        env:
+          BASE_REF: ${{ github.base_ref }}
         run: |
+          git config --global user.email "you@example.com"
+          git config --global user.name "Your Name"
+          git remote -v
```
Comment (asbjornsabo) [inline .github/workflows/bluetooth-tests.yaml:49]: Are these the name and email we want?

**Review by asbjornsabo — COMMENTED** [review]
Wondering about the name/email setting - otherwise this seems consistent with other scripts.

File: .github/workflows/bluetooth-tests.yaml (hunk 1 above)
Comment (carlescufi) [reply .github/workflows/bluetooth-tests.yaml:49]: this is just so that Git doesn't complain, it won't be used at all.

Comment (Thalley) [reply .github/workflows/bluetooth-tests.yaml:49]: I see that some of the other Github actions use 
```
          git config --global user.email "actions@zephyrproject.org"
          git config --global user.name "Github Actions"
          git remote -v
```

Comment (Thalley) [reply .github/workflows/bluetooth-tests.yaml:49]: There's also one using 
```
          git config --global user.email "bot@zephyrproject.org"
          git config --global user.name "Zephyr Bot"
```

Comment (asbjornsabo) [reply .github/workflows/bluetooth-tests.yaml:49]: Both of these alternatives seem cleaner to me.

**Review by gmarull — APPROVED** [review]
+1 because other CI workflows seem to do the same. However, I don't like the idea of doing “smart” rebases for our users. There's now the “Update branch” button that can be used if people suspect the checks need to be re-run.

File: .github/workflows/bluetooth-tests.yaml (hunk 1 above)
Comment (carlescufi) [reply .github/workflows/bluetooth-tests.yaml:49]: As I mentioned, we rebase here, this user config will never be actually used.

**Comment by Thalley** [comment]
> +1 because other CI workflows seem to do the same. However, I don't like the idea of doing “smart” rebases for our users. There's now the “Update branch” button that can be used if people suspect the checks need to be re-run.

The issue with rebasing, either via Github UI or force push, is that is (may) dismiss existing approvals. This way we can simply rerun the tests to run them on the newest commit, which, to me (and given the change to out GH settings), is a significant improvement.

**Review by asbjornsabo — APPROVED** [review]

**Review by Thalley — APPROVED** [review]

File: .github/workflows/bluetooth-tests.yaml [hunk 2]
```diff
@@ -41,7 +41,13 @@ jobs:
         uses: actions/checkout@v2
```
Comment (stephanosio) [inline .github/workflows/bluetooth-tests.yaml:41]: This does shallow clone and rebasing onto a different branch will not work.

> fatal: invalid upstream 'origin/main'

Need to specify `fetch-depth: 0`.

**Pull request merged**